src/Core/ObjectRegistry.php

```diff
@@ -123,23 +120,21 @@ protected function _checkDuplicate($name, $config)
         $existingConfig = $existing->config();
         unset($config['enabled'], $existingConfig['enabled']);
 
-        $fail = false;
         foreach ($config as $key => $value) {
             if (!array_key_exists($key, $existingConfig)) {
-                $fail = true;
-                break;
+                goto fail;
             }
             if (isset($existingConfig[$key]) && $existingConfig[$key] !== $value) {
-                $fail = true;
-                break;
+                goto fail;
             }
         }
-        if ($fail) {
-            $msg .= ' with the following config: ';
-            $msg .= var_export($existingConfig, true);
-            $msg .= ' which differs from ' . var_export($config, true);
-            throw new RuntimeException($msg);
-        }
+        return;
+
+        fail:
```

Comment:
haha :+1:... Not bad but what if you create the exception before hand and just `throw $exception` instead of `goto fail` ?